plugins/home-react/src/alpha/blueprints/CustomHomepageWidgetBlueprint.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { lazy, ReactElement } from 'react';
+import { compatWrapper } from '@backstage/core-compat-api';
+import { attachComponentData } from '@backstage/core-plugin-api';
+import {
+  coreExtensionData,
+  createExtensionBlueprint,
+  ExtensionBoundary,
+} from '@backstage/frontend-plugin-api';
+import {
+  CardConfig,
+  CardExtension,
+  CardExtensionProps,
+  CardLayout,
+  CardSettings,
+  ComponentParts,
+} from '../../extensions';
+
+/** @alpha */
+export interface CustomHomepageWidgetBlueprintParams {
+  /**
+   * Unique name for the widget. This value is exposed through component data
+   * and must match when referencing widgets from layout configuration.
+   */
+  name: string;
+  /**
+   * Optional title displayed for the widget, used as the default card heading.
+   */
+  title?: string;
+  /**
+   * Optional description shown in the widget catalog when adding new cards.
+   */
+  description?: string;
+  /**
+   * Component parts rendered within the card.
+   */
+  components: () => Promise<ComponentParts>;
+  /**
+   * Layout hints used by the customizable grid.
+   */
+  layout?: CardLayout;
+  /**
+   * Schema used to configure widget settings.
+   */
+  settings?: CardSettings;
+  /**
+   * Default props forwarded to the rendered widget component.
+   */
+  componentProps?: Record<string, unknown>;
+}
+
+const DEFAULT_WIDGET_ATTACH_POINT = {
+  id: 'home:custom-homepage:home',
+  input: 'widgets',
+} as const;
+
+/**
+ * Creates widgets that can be installed into the customizable home page grid.
+ *
+ * @alpha
+ */
+export const CustomHomepageWidgetBlueprint = createExtensionBlueprint({
+  kind: 'home:widget',
+  attachTo: DEFAULT_WIDGET_ATTACH_POINT,
+  output: [coreExtensionData.reactElement],
```

Comment:
I wonder if there's a case to be made for having a more complex set of outputs here. At least in the end. The [entity card blueprint](https://github.com/backstage/backstage/blob/7b41d9dd31e831c00f6c08f50090b9c1acdd94ce/plugins/catalog-react/src/alpha/blueprints/EntityCardBlueprint.ts#L45) for example has a `type` that it outputs that controls placement.

Do you have any ideas around how widget params can be designed, such that users are able to control the overall final presentation of the home page?